Modules/_io/bufferedio.c

```diff
@@ -240,8 +244,15 @@ typedef struct {
 
     PyObject *dict;
     PyObject *weakreflist;
+
+    /* a doubly-linked chained list of "buffered" objects that need to
+       be flushed when the process exits */
+    struct doubly_linked_s buffered_writers_list;
```

Comment:
The nested struct makes for quirky pointer calculation in _PyIO_atexit_flush(). Why not simply:
```
typedef struct _buffered {
    // ...
    struct _buffered *next, *prev;
} buffered;
```